pyro/poutine/subsample_messenger.py

```diff
@@ -47,7 +47,7 @@ def sample(self, sample_shape=torch.Size()):
             raise NotImplementedError
         subsample_size = self.subsample_size
         if subsample_size is None or subsample_size >= self.size:
-            result = jit_compatible_arange(self.size, device=self.device)
+            result = torch.arange(self.size, device=self.device)
```

Comment:
I've also removed our obsolete helper `jit_compatible_arange`, saving a few tensor ops in the most common case.